activity/mqtt/activity.go

```diff
@@ -89,7 +90,20 @@ func New(ctx activity.InitContext) (activity.Activity, error) {
 	if err != nil {
 		return nil, err
 	}
+	if settings.SharedConnection != "" {
+		
+		client, err := coerce.ToConnection(settings.SharedConnection)
+		if err != nil {
+			return nil, err
+		}
 
+		act := &Activity{
+			settings: settings,
+			client : client.GetConnection().(mqtt.Client),
```

Comment:
You probably want to keep a handle to the connection manager, and get a connection when you need it